### PR TITLE
Move main.go to cmd/ and split off config.go

### PR DESCRIPTION
**Pull request by faisal-memon**
Minor refactor of main.go:

1. Move `main.go` and `config_test.go` to `cmd/spire-controller-manager/`
2. Split off config related functions to `config.go`

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -28,7 +28,7 @@ COPY --link --from=xx / /
 RUN xx-go --wrap
 RUN --mount=type=cache,target=/root/.cache/go-build \
     --mount=type=cache,target=/go/pkg/mod \
-    go build -o bin/spire-controller-manager main.go
+    go build -o bin/spire-controller-manager ./cmd/spire-controller-manager
```
Comment (faisal-memon) [inline Dockerfile:31]: This should just be calling `make build`. I tried making that change but got this error. Its not as trivial as I thought:

```
 => ERROR [builder 4/4] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg/mod     make build                                                                                                                                                                                                       0.1s
------
 > [builder 4/4] RUN --mount=type=cache,target=/root/.cache/go-build     --mount=type=cache,target=/go/pkg/mod     make build:
#0 0.063 env: can't execute 'bash': No such file or directory
#0 0.064 env: can't execute 'bash': No such file or directory
#0 0.065 env: can't execute 'bash': No such file or directory
#0 0.065 Makefile:58: *** unsupported OS: .  Stop.
------
Dockerfile:31
--------------------
  30 |     RUN apk add --update make
  31 | >>> RUN --mount=type=cache,target=/root/.cache/go-build \
  32 | >>>     --mount=type=cache,target=/go/pkg/mod \
  33 | >>>     make build
  34 |         #go build -o bin/spire-controller-manager ./cmd/spire-controller-manager
--------------------
ERROR: failed to solve: process "/bin/sh -c make build" did not complete successfully: exit code: 2
```

**Comment by azdagron** [comment]
Is this just a code shuffle? Is there any other benefits introduced by this change? 
The structure of main.go, the code layout, and the Makefile are not my favorite but departures from the project code layout impacts our ability to adjust to newer versions of kubebuilder, which we've undertaken on occasion.

I'd rather not take this change unless there is a very clear benefit that outweighs the pain that it will introduce.

**Comment by faisal-memon** [comment]
No changes, just reorganization.

**Comment by azdagron** [comment]
Cool. Thanks for taking the time to spread the code cleanliness love even though we ultimately aren't taking it.

**Pull request closed without merging**